CMakeLists.txt

```diff
@@ -65,6 +65,16 @@ else()
             Subversion_WC_INFO(${PROJ_ROOT} Project)
             add_definitions(-DPADDLE_VERSION=${Project_WC_REVISION})
         endif()
+    elseif(EXISTS ${PROJ_ROOT}/.git/)
+        find_package(Git REQUIRED)
+        if(GIT_FOUND)
+            execute_process(
+                COMMAND git log -1 --format=%H
+                WORKING_DIRECTORY ${PROJ_ROOT}
+                OUTPUT_VARIABLE GIT_SHA1
+                ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
check return value for `git log` command.  https://cmake.org/cmake/help/v3.0/command/execute_process.html
```
execute_process(...
RESULT_VARIABLE GIT_RETURN
...
)

if (${GIT_RETURN} != 0) {
    log(fatal) << "show error log".
}
```